.github/workflows/release.yml

```diff
@@ -63,48 +72,53 @@ jobs:
            </servers>
           </settings>
           EOF
+
       - name: Build with Maven
         run: |
           mvn versions:set -DremoveSnapshot versions:commit
           mvn -P dse -q -ff clean package -DskipTests
-      - name: zip-up
+
+      - name: Zip-up `stargate-lib`
         run: |
           zip stargate-jars.zip starctl stargate-lib/logback.xml stargate-lib/*.jar
-      - name: Retrieve stashed release URL
-        uses: actions/download-artifact@v1
-        with:
-          name: release-url
-      - name: Read release URL
-        id: get_release_url
-        run: echo ::set-output name=URL::$(cat release-url/url.txt)
+          
+      # uploads the jars by referencing the release-upload-url from create-release job
       - name: Upload jars
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.get_release_url.outputs.URL }}
+          upload_url: ${{needs.create-release.outputs.release-upload-url}}
```

Comment:
Should this be `steps.create-release...` instead of `needs.create-release...`?

Reply:
This uses the referenced job in the `needs` section, and then uses it's outputs: https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs